src/context/WalletProvider/MetaMask/components/Connect.tsx

```diff
@@ -36,17 +36,15 @@ export const MetaMaskConnect = ({ history }: MetaMaskSetupProps) => {
     setLoading(false)
   }, [])
 
-  useEffect(() => {
-    void onProviderChange(KeyManager.MetaMask)
-  }, [onProviderChange])
```

Comment:
This was way too reacty and produced multiple MM JSON-RPC requests, v. bad. 
Made `onProviderChange` actually return the provider so it can be used in callbacks in a non-reactive fashion.

Reply:
Note, all other MM-like providers consuming `onProviderChange()` will need to be refactored as well